### PR TITLE
[BUGFIX] Fix restore panic in case of volumesnapshot missing

### PR DESCRIPTION
**Pull request by ShellyKa13**
### What this PR does
Before this PR:
In case of volumesnapshot missing while in the restore process we are getting to a nil pointer dereference

After this PR:
Handle nil pointer and return error that volumesnapshot missing

Fixes # jira-ticket: https://issues.redhat.com/browse/CNV-42228

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
BugFix: Fix restore panic in case of volumesnapshot missing
```

### CODE REVIEW

File: pkg/storage/snapshot/restore.go [hunk 1]
```diff
@@ -968,11 +956,17 @@ func (ctrl *VMRestoreController) createRestorePVC(
 	if err != nil {
 		return err
 	}
+	if volumeSnapshot == nil {
```
Comment (akalenyu) [inline pkg/storage/snapshot/restore.go:959]: @mhenriks Do you think this is physically possible without malicious-like actions by the user?

**Review by akalenyu — APPROVED** [review]

File: pkg/storage/snapshot/restore.go (hunk 1 above)
Comment (dankenigsberg) [reply pkg/storage/snapshot/restore.go:959]: imho we should ensure that https://github.com/kubevirt/kubevirt/blob/356f5376b7e990ef105c4dbbf385c1f56e26a37d/pkg/storage/snapshot/snapshot_base.go#L668 never return (`nil`, `nil`)

Comment (mhenriks) [reply pkg/storage/snapshot/restore.go:959]: @dankenigsberg It is a very common pattern in this project for functions that are wrappers around informers/indexes to return `nil, nil` if the resource does not exist.

Comment (mhenriks) [reply pkg/storage/snapshot/restore.go:959]: @akalenyu it is possible that VolumeSNapshots are deleted by admins for malicious reasons probably to free up storage on the back end

Comment (dankenigsberg) [reply pkg/storage/snapshot/restore.go:959]: @mhenriks thanks. Is it a kubevirt thing, or is it commonplace in other k8s controllers?

I (and probably the author of the modified code) find it it very surprising for a getter function to get nothing yet declare success.

**Pull request merged**